torch/distributed/tensor/_sharding_prop.py

```diff
@@ -198,8 +199,25 @@ def _propagate_tensor_meta_non_cached(
     def _propagate_tensor_meta(
         self, op_schema: OpSchema
     ) -> Union[None, TensorMeta, Sequence[Optional[TensorMeta]]]:
+        """
+        Cached version of _propagate_tensor_meta_non_cached
+        This is a private API. Use propagate_tensor_meta instead.
+        """
         return self._propagate_tensor_meta_non_cached(op_schema)
 
+    def propagate_tensor_meta(
```

Comment:
hmm @XilunWu is it an intentional decision in DTensor to have the existing `_propagate_tensor_meta` methods be private? @azahed98 if so we should probably keep things that way (maybe just rename `_propagate_tensor_meta` to `_propagate_tensor_meta_cached`)